design/FY2016/CondFDNodeFluxOutput.md

```diff
@@ -0,0 +1,120 @@
+Conduction Finite Difference Node Heat Flux Reporting
+================
+
+**M.J. Witte, GARD Analytics, Inc.**
+
+ - October 30, 2015
+ - *Revision Date*
+ 
+
+## Justification for New Feature ##
+
+The primary motivation for this work is to support experimental testing where heat flux meters are embedded within a wall. It is also useful for other work to show the flow of heat within material layers, especially for phase change material (PCM) applications.
+
+## E-mail and  Conference Call Conclusions ##
+
+*insert text*
+
+## Overview ##
+
+Currently, the conduction finite difference algorithm (HeatBalanceAlgorithm,ConductionFiniteDifference) reports the temperature at each node within a surface.  This will add reporting of the heat flux between nodes.
+
+## Approach ##
+
+At the end of the conduction finite difference (CondFD) solution for each timestep, the surface inside face conduction heat flux and the node temperatures will be used to calculate the heat flux between nodes.  The figure below illustrates the conduction finite difference node layout (from the Engineering Reference).
+
+
+![CondFD Nodes adapted from Engineering Reference](CondFDNodeFluxOutput-CondFDNodes-image176.png)
+
+Figure 1. Node depiction for Conduction Finite Difference Model (Nodes are numbered from outside to outside face to inside face.  External surface node = 1, Interior surface node = 8)
+
+For every CondFD surface, the Surface Inside Face Conduction Heat Transfer Rate per Area is available as an output. Starting with this flux, the heat flux from the Interior surface node 8 to the next interior node 7 is the Surface Inside Face Conduction Heat Transfer Rate per Area plus any change in heat storage in the material thickness associated with node 8.  Once the flux from node 8 to node 7 is determined, then the next flux can be calculated, and so on.
```

Comment:
I'm probably oversimplifying things a bit, but my understanding from the various node calc routines was that the only type of node with two half-node heat capacities are the nodes that sit at the interface between two materials.  The other calc routines only compute a single Cp, Rho, and delX.  That only shows in the code changes below (look for cp1 or cp2).  And if one of the materials is an r-layer, then it's cp is zero so only cp1 or only cp2 gets used.